src/PathOrderMonotonic.h

```diff
@@ -0,0 +1,449 @@
+     * \param line_bucket_grid A pre-computed bucket grid to allow quick look-up
+     * of which vertices are nearby.
+     * \return A list of polylines, in the order in which they should be
+     * printed. All paths in this string already have their start_vertex set
+     * correctly.
+     */
+    std::deque<Path*> findPolylineString(Path* polyline, const SparsePointGridInclusive<Path*>& line_bucket_grid, const Point monotonic_vector)
+    {
+        std::deque<Path*> result;
+        if(polyline->converted->empty())
+        {
+            return result;
+        }
+
+        //Find the two endpoints of the polyline string, on either side.
+        result.push_back(polyline);
+        polyline->start_vertex = 0;
+        Point first_endpoint = polyline->converted->front();
+        Point last_endpoint = polyline->converted->back();
+        std::vector<SparsePointGridInclusiveImpl::SparsePointGridInclusiveElem<Path*>> lines_before = line_bucket_grid.getNearby(first_endpoint, COINCIDENT_POINT_DISTANCE);
+        auto close_line_before = std::find_if(lines_before.begin(), lines_before.end(), [first_endpoint, result](SparsePointGridInclusiveImpl::SparsePointGridInclusiveElem<Path*> found_path) {
+            return found_path.val->start_vertex == found_path.val->converted->size() //Don't find any line already in the string.
+                   && vSize2(found_path.point - first_endpoint) < SQUARED_COINCIDENT_POINT_DISTANCE; //And only find close lines.
+        });
+        std::vector<SparsePointGridInclusiveImpl::SparsePointGridInclusiveElem<Path*>> lines_after = line_bucket_grid.getNearby(last_endpoint, COINCIDENT_POINT_DISTANCE);
+        auto close_line_after = std::find_if(lines_after.begin(), lines_after.end(), [last_endpoint, result](SparsePointGridInclusiveImpl::SparsePointGridInclusiveElem<Path*> found_path) {
+            return found_path.val->start_vertex == found_path.val->converted->size() //Don't find any line already in the string.
+                   && vSize2(found_path.point - last_endpoint) < SQUARED_COINCIDENT_POINT_DISTANCE; //And only find close lines.
+        });
+
+        while(close_line_before != lines_before.end())
+        {
+            Path* first = close_line_before->val;
+            result.push_front(first); //Store this one in the sequence. It's a good one.
+            size_t farthest_vertex = getFarthestEndpoint(first, close_line_before->point); //Get to the opposite side.
+            first->start_vertex = farthest_vertex;
+            first->backwards = farthest_vertex != 0;
+            first_endpoint = (*first->converted)[farthest_vertex];
+            lines_before = line_bucket_grid.getNearby(first_endpoint, COINCIDENT_POINT_DISTANCE);
+            close_line_before = std::find_if(lines_before.begin(), lines_before.end(), [first_endpoint, result](SparsePointGridInclusiveImpl::SparsePointGridInclusiveElem<Path*> found_path) {
```

Comment:
So a couple of things about these specific inline closures.
- I don't think the capture of 'result' is necessary here.
- Propose: If you capture the other one (`..._endpoint`) as a reference or pointer instead, you could make one (named) closure, set the reference, and then feed the same one to the `.find(...)` each time. (Or make an actual lambda or function, and feed the reference via a `std::bind`.)

(Related: Perhaps the two earlier blocks could also be folded into the while loops, but let's not go overboard here.)

Reply:
Yeah I considered factoring this repeated bit of code out. I think I'll do that. It's even caused bugs for me before because I forgot to update the two calls above the loop.

The `result` capture is indeed a leftover from a previous implementation. A bugfix on Tuesday to fix those infinite loops made this bit a lot simpler and removed the need to capture `result`.

I'm opting for the second option, feeding the point as an argument to an actual function. Storing the two points in the same variable is a bad idea because they are both used below (which needs the old value of the front still). That could be resolved too by interweaving that code, but I think this is clearer. And if it's a named function you'll have to give it that argument anyway.

Since `std::bind_front` is not available until C++20, I'll just have to use a lambda.

Done here: 47fe5b9e5a520de0682edad044ab79a70a281861

Reply:
Yes, this is what I wanted. (I have a hard time keeping track of what's in C++17 versus what's in C++20 for some reason.)